includes/class-wc-product-variable.php

```diff
@@ -455,11 +455,15 @@ public static function sync( $product, $save = true ) {
 			$data_store->sync_stock_status( $product );
 			self::sync_attributes( $product ); // Legacy update of attributes.
 
-			do_action( 'woocommerce_variable_product_sync', $product->get_id(), $product->get_visible_children(), $save );
+			do_action( 'woocommerce_variable_product_sync_before_save', $product );
 
 			if ( $save ) {
 				$product->save();
 			}
+
+			do_action( 'woocommerce_variable_product_synced', $product );
```

Comment:
I don't agree that this one is needed. Unless `$save` is true, this hook is identical to the one above, and if you really did need to kick logic in at this point there are action hooks in each object's save/create/update methods which can be used instead so you can capture all save events.

Did I miss a use case?